lib/licensed/sources/nuget.rb

```diff
@@ -0,0 +1,226 @@
+# frozen_string_literal: true
+require "json"
+
+module Licensed
+  module Sources
+    # Only supports ProjectReference (project.assets.json) style restore used in .NET Core.
+    # Does not currently support packages.config style restore.
+    class NuGet < Source
+      def self.type
+        "nuget"
+      end
+
+      # Match nuspec license expressions like <license type="expression">MIT</license>
+      class NuspecLicenseExpressionMatcher < Licensee::Matchers::Package
+        LICENSE_EXPRESSION_REGEX = /
+          <license\s*type\s*=\s*\"\s*expression\s*\"\s*>\s*([a-z\-0-9\.]+)\s*<\/license>
+        /ix.freeze
+
+        def license_property
+          return @license_property if defined?(@license_property)
+          match = @file.content.match LICENSE_EXPRESSION_REGEX
+          if match && match[1]
+            @license_property = match[1].downcase
+          end
+        end
+      end
+
+      class NuspecFile < Licensee::ProjectFiles::ProjectFile
+        def possible_matchers
+          [NuspecLicenseExpressionMatcher]
+        end
+      end
+
+      # A pseudo-license file where the license is already known (e.g. inferred from a URL)
+      class KnownLicenseFile < Licensee::ProjectFiles::ProjectFile
```

Comment:
👍 this sounds helpful.  since it's generic and not tied to this class, can be moved to its own file like `lib/licensed/project_files/known_license_file.rb`?

 or maybe licensee would be ok with having this class as an official licensee project file type?